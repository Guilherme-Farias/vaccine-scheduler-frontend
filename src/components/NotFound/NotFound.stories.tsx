import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import NotFound from '.';

export default {
  title: 'NotFound',
  component: NotFound,
} as ComponentMeta<typeof NotFound>;

const Template: ComponentStory<typeof NotFound> = args => (
  <NotFound {...args} />
);

export const Default = Template.bind({});
Default.args = {};
