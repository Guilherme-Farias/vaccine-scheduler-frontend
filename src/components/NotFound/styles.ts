import styled, { css } from 'styled-components/macro';

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    gap: ${theme.spacings.lg};
    height: 100%;

    @media ${theme.queries.tabletAndSmaller} {
      justify-content: start;
      flex-direction: column-reverse;
      align-items: center;
      gap: ${theme.spacings.sm};
    }
  `}
`;

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes['5xl']};

    @media ${theme.queries.tabletAndSmaller} {
      font-size: ${theme.font.sizes['3xl']};
    }
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.lg};
  `}
`;

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    max-width: 300px;

    margin-top: ${theme.spacings['6xl']};
    @media ${theme.queries.tabletAndSmaller} {
      max-width: revert;
      margin-top: ${theme.spacings['2xl']};
    }
  `}
`;

export const Img = styled.img`
  max-width: 628px;
`;
