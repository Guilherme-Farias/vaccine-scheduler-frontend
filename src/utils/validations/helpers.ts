import Joi from 'joi';
import { IParamErrors } from 'types/global';

export function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: IParamErrors = {};

  if (objError.error) {
    objError.error.details.forEach(err => {
      errors[err.path.join('.')] = err.message;
    });
  }

  return errors;
}
