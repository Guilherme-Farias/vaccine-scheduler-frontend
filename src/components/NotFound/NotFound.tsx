import React from 'react';
import { useNavigate } from 'react-router-dom';

import MaxWidthWrapper from 'components/MaxWidthWrapper';
import Button from 'components/Button';

import notFoundImg from 'assets/images/not-found.svg';
import * as S from './styles';

const NotFound: React.FC = () => {
  const navigate = useNavigate();
  return (
    <MaxWidthWrapper>
      <S.Content>
        <section>
          <S.Title>Algo não está certo...</S.Title>
          <S.Description>
            A página que você está tentando abrir não existe. Você pode ter
            digitado incorretamente o endereço ou a página foi movido para outra
            URL. Se você acha que isso é um erro, entre em contato com o
            suporte.
          </S.Description>
          <S.ButtonContainer>
            <Button variant="outline" fullWidth onClick={() => navigate('/')}>
              Voltar para a página inicial
            </Button>
          </S.ButtonContainer>
        </section>
        <S.Img
          src={notFoundImg}
          alt="Uma mulher procurando por algo e não achando. O fundo da imagem é número 404"
          title="Ilustração do error 404 no servidor (Não encontrado), vetor criado por Storyset - www.freepik.com/vectors/server-error"
        />
      </S.Content>
    </MaxWidthWrapper>
  );
};
export default NotFound;
