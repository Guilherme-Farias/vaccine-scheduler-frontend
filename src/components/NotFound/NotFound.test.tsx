import { render, screen, userEvent } from 'utils/tests/helpers';

import NotFound from '.';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('<NotFound />', () => {
  it('should render NotFound ', () => {
    const { container } = render(<NotFound />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should navigate to "/" (Home)', async () => {
    render(<NotFound />);

    await userEvent.click(
      screen.getByRole('button', { name: /voltar para a p√°gina inicial/i }),
    );

    expect(mockNavigate).toHaveBeenCalledWith('/');
  });
});
