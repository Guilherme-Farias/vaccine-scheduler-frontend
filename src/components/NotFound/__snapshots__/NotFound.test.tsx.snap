// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render NotFound  1`] = `
.c0 {
  width: 100%;
  max-width: 81.25rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
}

.c5 {
  border: 2px solid transparent;
  border-radius: 0.625rem;
  font-weight: 500;
  font-size: 1.125rem;
  padding: 0.75rem 1.25rem;
  width: 100%;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c5:focus {
  outline-color: #D03E35;
  outline-offset: 4px;
}

.c6 {
  background-color: transparent;
  color: #D03E35;
  border: 2px solid currentColor;
}

.c6:hover {
  background-color: rgba(208,62,53,0.1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.25rem;
  height: 100%;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 1.125rem;
}

.c4 {
  max-width: 300px;
  margin-top: 3rem;
}

.c7 {
  max-width: 628px;
}

@media (max-width:68.75rem) {
  .c1 {
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 0.75rem;
  }
}

@media (max-width:68.75rem) {
  .c2 {
    font-size: 1.875rem;
  }
}

@media (max-width:68.75rem) {
  .c4 {
    max-width: revert;
    margin-top: 1.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <section>
      <h2
        class="c2"
      >
        Algo não está certo...
      </h2>
      <p
        class="c3"
      >
        A página que você está tentando abrir não existe. Você pode ter digitado incorretamente o endereço ou a página foi movido para outra URL. Se você acha que isso é um erro, entre em contato com o suporte.
      </p>
      <div
        class="c4"
      >
        <button
          class="c5 c6"
          color="primary"
        >
          Voltar para a página inicial
        </button>
      </div>
    </section>
    <img
      alt="Uma mulher procurando por algo e não achando. O fundo da imagem é número 404"
      class="c7"
      src="not-found.svg"
      title="Ilustração do error 404 no servidor (Não encontrado), vetor criado por Storyset - www.freepik.com/vectors/server-error"
    />
  </div>
</div>
`;
